src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/TestCaseCachingService.kt

```diff
@@ -0,0 +1,187 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.services
+
+import com.intellij.openapi.diagnostic.Logger
+import org.evosuite.utils.CompactReport
+import org.evosuite.utils.CompactTestCase
+
+class TestCaseCachingService {
+    private val log: Logger = Logger.getInstance(this.javaClass)
+    private val files = mutableMapOf<String, FileTestCaseCache>()
+    private val filesLock = Object()
+
+    /**
+     * Insert test cases into the cache.
+     *
+     * @param fileUrl the URL of the file that the test cases belong to
```

Comment:
What is the file url? Is this a file path or something? Where can I get the URL for the file? Thank you.